src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Options/EditorConfigValueSerializer.cs

```diff
@@ -17,7 +18,7 @@ private static string UnescapeLineBreaks(string str)
             => str.Replace("\r", "\\r").Replace("\n", "\\n");
 
         private static readonly EditorConfigValueSerializer<bool> s_bool = new(
-            parseValue: str => bool.TryParse(str, out var result) ? result : new Optional<bool>(),
+            parseValue: ParseBoolean,
             serializeValue: value => value ? "true" : "false");
```

Comment:
>value ? "true" : "false" [](http://example.com/codeflow?start=37&length=24)

You can also extract this to a helper method and use it in `s_nullableBoolean`. Then you could also make the lambda there a simple one liner.